server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -924,7 +925,8 @@ private void getAssignmentsFromBalancer(TabletLists tLists,
                   lastLocation.getHostPort());
             }
 
-            tLists.assignments.add(new Assignment(assignment.getKey(), assignment.getValue()));
+            tLists.assignments.add(new Assignment(assignment.getKey(), assignment.getValue(),
+                lastLocation != null ? Location.last(lastLocation) : null));
```

Comment:
Looking around in the code I am not entirely sure `lastLocation` is actually a location from the last column. Not sure what the value in the map is for ATM.  Entries are added to the unassigned map in the following method and that does not look like its adding things from the last column to the map.

https://github.com/apache/accumulo/blob/9c459ce8f06d9d9e04280d020ea091fedf13749b/server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java#L383-L399

Reply:
I updated the PR to use the lastLocation value that is part of TabletLocationState when doing the metadata updates. Before the change the value was always coming from getServer() which is returning getLocation() and that could be current, future or last so this makes sure to always use the last value from metadata. The old value (getServer()) is still being tracked now as I created an UnassignedTablet class to track both that value and lastLocation as they are used for different things.